rir/src/ir/Compiler.cpp

```diff
@@ -277,8 +278,21 @@ bool compileSimpleFor(CompilerContext& ctx, SEXP fullAst, SEXP sym, SEXP seq,
             cs << BC::colonInputEffects();
             cs.addSrc(seq);
             bool staticFastcase =
-                (isConstant(start) && !inherits(start, "factor")) ||
-                (isConstant(end) && !inherits(end, "factor"));
+                (isConstant(start) && !inherits(start, "factor") &&
```

Comment:
ok,I am reading this code for the 3rd time, still no clue what it does, and each time it looks a bit different...